packages/ui/src/components/Designer/index.tsx

```diff
@@ -113,6 +113,8 @@ const TemplateEditor = ({
           )?.propPanel;
           set(tgt, 'data', propPanel?.defaultValue || '');
           Object.assign(tgt, propPanel?.defaultSchema || {});
+        } else if (key === 'data' && tgt.readOnly) {
```

Comment:
instead of this, would it be better to provide an alternative function to be passed into the renderers?

We have this `onChange` function for all schemas which represents when the `data` changes. 
For a flexible plugin API we could add a `onCustomAttributeChange` function, which allows the plugin to trigger a change on whichever attribute they like?

Reply:
(I need something like this for my plugin :D )